beta-src/src/components/ui/WDMain.tsx

```diff
@@ -253,6 +258,78 @@ const WDMain: React.FC = function (): React.ReactElement {
       isLivePhase: false,
     };
   };
+
+  useEffect(() => {
+    Object.entries(ordersMeta).forEach(([_, orderMeta]) => {
+      // Let's use as an example ION C APU - TUN
+      // The board should automatically also enters APU - TUN (Via Convoy)
+      // More info in CHR-68
+      const orderMetaUpdate = orderMeta.update;
+
+      if (
+        orderMetaUpdate?.type === "Convoy" &&
+        orderMetaUpdate?.fromTerrID &&
+        orderMetaUpdate.toTerrID
+      ) {
+        const fromTerr = maps.terrIDToProvince[orderMetaUpdate.fromTerrID];
+        const toTerr = maps.terrIDToProvince[orderMetaUpdate.toTerrID];
+
+        const legalConvoysByUnitIDs = Object.keys(
+          legalOrders.hasAnyLegalConvoysByUnitID,
+        );
+        if (legalConvoysByUnitIDs.length === 1) {
+          const availableViaConvoy = legalConvoysByUnitIDs[0];
+          // Verify if the order APU -> TUN is possible
+          const legalOrderViaConvoy =
+            legalOrders.legalConvoysByUnitID[availableViaConvoy];
+          if (legalOrderViaConvoy[fromTerr][toTerr]) {
```

Comment:
I forget the corners of how javascript/typescript work. Will this barf if the outer map key is absent, so that you end up doing something like undefined[toTerr] when you then proceed to try to apply the inner map key?

Reply:
it will never get to undefined[toTerr] because there is a previous validation `legalConvoysByUnitIDs.length === 1`

Reply:
condition removed